### PR TITLE
Feature: Market Feed Service

### PR DESCRIPTION
**Pull request by SarumatheK**
Whats Changed:
- Added Code to get new data from Alpha Vantage API for rendering to frontend

### CODE REVIEW

File: src/fin_webapp/Gemfile [hunk 1]
```diff
@@ -0,0 +1,53 @@
+source "https://rubygems.org"
+
+ruby "3.3.0"
+
+# Bundle edge Rails instead: gem "rails", github: "rails/rails", branch: "main"
+gem "rails", "~> 7.1.3"
+
+# Use sqlite3 as the database for Active Record
+gem "sqlite3", "~> 1.4"
```
Comment (rishirajkumar97) [inline src/fin_webapp/Gemfile:9]: remove this gem as its not required and you dont use any database

File: src/fin_webapp/app/controllers/market_data_controller.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class MarketDataController < ApplicationController
```
Comment (rishirajkumar97) [inline src/fin_webapp/app/controllers/market_data_controller.rb:1]: nit: marketNewsData

File: src/fin_webapp/app/controllers/market_data_controller.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+  def index
+    alpha_vantage_service = AlphaVantageService.new
+    @market_data = alpha_vantage_service.get_news_sentiment(params[:symbol])
```
Comment (rishirajkumar97) [inline src/fin_webapp/app/controllers/market_data_controller.rb:4]: check if params is present or throw an error if params[:symbol] is empty

Comment (rishirajkumar97) [inline src/fin_webapp/app/controllers/market_data_controller.rb:4]: why use it as @market_data I dont think its required to be a global variable , replace without the '@'

File: src/fin_webapp/app/services/alpha_vantage_service.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# app/services/alpha_vantage_service.rb
+
+require 'faraday'
+require 'json'
+
+class AlphaVantageService
+  def initialize
+    @api_key = ENV['ALPHA_VANTAGE_API_KEY']
+    @connection = Faraday.new(url: 'https://www.alphavantage.co')
```
Comment (rishirajkumar97) [inline src/fin_webapp/app/services/alpha_vantage_service.rb:9]: convert the URL to a constant and store it

File: src/fin_webapp/app/services/alpha_vantage_service.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+require 'faraday'
+require 'json'
+
+class AlphaVantageService
+  def initialize
+    @api_key = ENV['ALPHA_VANTAGE_API_KEY']
+    @connection = Faraday.new(url: 'https://www.alphavantage.co')
+  end
+
+  def get_news_sentiment(symbol, limit = 1000)
```
Comment (rishirajkumar97) [inline src/fin_webapp/app/services/alpha_vantage_service.rb:12]: same for the 1000 store in a constant

**Review by rishirajkumar97 — COMMENTED** [review]
please do address comments

File: src/fin_webapp/app/controllers/market_data_controller.rb (hunk 3 above)
Comment (SarumatheK) [reply src/fin_webapp/app/controllers/market_data_controller.rb]: 63cae27 resolved

File: src/fin_webapp/app/controllers/market_data_controller.rb (hunk 2 above)
Comment (SarumatheK) [reply src/fin_webapp/app/controllers/market_data_controller.rb]: resolved 63cae27

File: src/fin_webapp/app/services/alpha_vantage_service.rb (hunk 4 above)
Comment (SarumatheK) [reply src/fin_webapp/app/services/alpha_vantage_service.rb]: 63cae27 resolved

File: src/fin_webapp/app/services/alpha_vantage_service.rb (hunk 5 above)
Comment (SarumatheK) [reply src/fin_webapp/app/services/alpha_vantage_service.rb]: resolved 63cae27

File: src/fin_webapp/app/services/alpha_vantage_service.rb [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    handle_response(response)
+  end
+
+  private
+
+  def handle_response(response)
+    if response.status == 200
+      JSON.parse(response.body)
+    else
+      { error: "Error fetching data from Alpha Vantage API: Status #{response.status}" }
```
[the remark this answers is not in the document]
Comment (SarumatheK) [reply src/fin_webapp/app/services/alpha_vantage_service.rb]: resolved 63cae27

File: src/fin_webapp/Gemfile (hunk 1 above)
Comment (rishirajkumar97) [reply src/fin_webapp/Gemfile:9]: parking for laters as require rails/all is not working as intented and might need more time to figure whats happening underneath the hood in Rails 7.

**Pull request merged**